Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -66,6 +67,14 @@ public class SearchSkimmer : Skimmer<AnalyzeContext>
                 string matchExpressionMessage = matchExpression.Message;
                 matchExpression.ArgumentNameToIndexMap = GenerateIndicesForNamedArguments(ref matchExpressionMessage);
 
+                if (!string.IsNullOrEmpty(matchExpression.DeprecatedName) && _deprecatedNames == null)
+                {
+                    _deprecatedNames = new List<string>
```

Comment:
>_deprecatedNames [](http://example.com/codeflow?start=20&length=16)

This data and the match expressions should be persisted to the context object passed to Analyze. We can take this change for now, but should fix this moving forwarqd.

Reply:
I don't think we should add this to the context. and the reason for that is:

today we have:
1 skimmer with everything and  1.000 contexts, with that, only one list of matchExpressions and one list of deprecatedNames.

if we change:
1 skimmer and 1.000 contexts, with that,  each context would have the list of matchExpressions and deprecatedNames.

our memory consumption would increase a lot...

Reply:
I think things/properties that do not change should be  in the skimmer and things that might change should be in the context.

for those cases, none would be in the context.